Hunting Queries/MultipleDataSources/AzureResourceCreationWithNetworkActivity.yaml

```diff
@@ -0,0 +1,99 @@
+id: ac25d05d-362d-4a8d-b4e7-58c0edd2379c
+name: Anomalous Resource Creation and related Network Activity
+description: |
+  'Indicates when an anomalous number of resources are created successfully in Azure via the AzureActivity log.
+  This is then joined with the AzureNetworkAnalytics_CL data to identify any network related activity for the created resource.
+  The anomaly detection identifies activities that have occured both since the start of the day 1 day ago and the start of the day 7 days ago.
+  The start of the day is considered 12am UTC time.
+  Resources: 
+  https://docs.microsoft.com/en-us/azure/azure-monitor/insights/azure-networking-analytics
+  https://docs.microsoft.com/en-us/azure/network-watcher/traffic-analytics-schema'
+requiredDataConnectors:
+  - connectorId: AzureActivity
+    dataTypes:
+      - AzureActivity
+  - connectorId: AzureNetworkWatcher
+    dataTypes:
+      - AzureNetworkAnalytics_CL
+tactics:
+  - Impact
+relevantTechniques:
+  - T1496
+query: |
+
+  let starttime = 7d;
+  let endtime = 1d;
+  let Activity = AzureActivity
+  | where TimeGenerated >= startofday(ago(starttime))
+  // We look for any Operation that created and then succeeded where ActivitySubStatus has a value so that we can provide context
+  | where OperationName has "Create"
+  | where ActivityStatus has "Succeeded"
+  | make-series dResourceCount=dcount(ResourceId) default=0 on EventSubmissionTimestamp in range(startofday(ago(7d)), now(), 1d) by Caller, Resource, OperationName
+  | extend (RSquare,Slope,Variance,RVariance,Interception,LineFit)=series_fit_line(dResourceCount)
+  // Comment slope reference below to see all returns
+  //| where Slope > 0.2
+  | join kind=leftsemi (
+  // Last day's activity is anomalous
+  AzureActivity
+  | where TimeGenerated >= startofday(ago(endtime))
+  // We look for any Operation that created and then succeeded where ActivitySubStatus has a value so that we can provide context
+  | where OperationName has "Create"
```

Comment:
Do you want to filter out resources you know will definitely create network activity such as network interface creation, availability set changes or LB updates? The anomaly detection might mean these are not an issue but I could see it potentially FPing.

Reply:
This is hunting, so again we are building context for a potential investigation.